src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -587,6 +588,83 @@ int sd_dhcp6_lease_get_captive_portal(sd_dhcp6_lease *lease, const char **ret) {
         return 0;
 }
 
+int dhcp6_lease_get_vendor_option(sd_dhcp6_lease *lease) {
+        int r;
+        sd_dhcp6_option **options = NULL;
+        size_t n_vendor_options;
+
+        assert(lease);
+
+        r = set_dump_sorted(lease->vendor_options, (void***)&options, &n_vendor_options);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+static int dhcp6_lease_insert_vendor_option(
+                sd_dhcp6_lease *lease,
+                uint16_t option_code,
+                const void *data,
+                size_t len,
+                uint16_t enterprise_id) {
+        _cleanup_(sd_dhcp6_option_unrefp) sd_dhcp6_option *option = NULL;
+        int r;
+
+        assert(lease);
+
+        option = new(sd_dhcp6_option, 1);
+        if (!option)
+                return -ENOMEM;
+
+        *option = (sd_dhcp6_option) {
+                .n_ref = 1,
+                .enterprise_identifier = enterprise_id,
+                .option = option_code,
+                .length = len,
+        };
+        option->data = memdup_suffix0(data, len);
+        if (!option->data) {
+                return -ENOMEM;
+        }
+
+        r = set_ensure_put(&lease->vendor_options, &dhcp6_option_hash_ops, TAKE_PTR(option));
+        if (r < 0)
+                return r;
+
+        sd_dhcp6_option_ref(TAKE_PTR(option));
+
+        return 1;
```

Comment:
Please use `_consume()`, otherwise `option` will be leaked on failure.
And the explicit `_ref()` is not necessary when it is consumed.
```c
return set_ensure_consume(&lease->vendor_options, &dhcp6_option_hash_ops, TAKE_PTR(option));
```

Reply:
Changed to use set_ensure_consume()